solutions/silver/usaco-860.mdx

```diff
@@ -130,200 +147,235 @@ import java.io.*;
-			if (largeRegions.size() == 0) { break; }
+		return cells;
+	}
 
-			// Delete squares that are part of a large region
-			for (int i = 0; i < n; i++) {
-				for (int j = 0; j < m; j++) {
-					if (largeRegions.contains(region[i][j])) { grid[i][j] = 0; }
-				}
-			}
+	static List<List<int[]>> findRemovableRegions(char[][] grid, int k) {
+		int n = grid.length;
+		List<List<int[]>> removableRegions = new ArrayList<>();
+		boolean[][] visited = new boolean[n][MAX_COLS];
 
-			/*
-			 * Apply gravity; In other words, for each column, we move all the
-			 * zeros to the top
-			 * Here's a good source explaining a similar concept:
-			 * https://www.geeksforgeeks.org/move-zeroes-end-array/
-			 */
-			for (int j = 0; j < m; j++) {
-				int count = n - 1;
-				for (int i = n - 1; i >= 0; i--) {
-					if (grid[i][j] != 0) { grid[count--][j] = grid[i][j]; }
-				}
+		for (int row = 0; row < n; row++) {
+			for (int col = 0; col < MAX_COLS; col++) {
+				// Skip already visited cells or empty cells ('0')
+				if (grid[row][col] == '0' || visited[row][col]) { continue; }
+
+				char color = grid[row][col];
+				List<int[]> cells = findRegion(grid, row, col);
+
+				// Mark all cells in the region as visited
+				for (int[] cell : cells) { visited[cell[0]][cell[1]] = true; }
 
-				while (count >= 0) { grid[count--][j] = 0; }
+				// If the region size meets the threshold, add it to the removable
+				// regions
+				if (cells.size() >= k) { removableRegions.add(cells); }
 			}
 		}
 
-		for (int i = 0; i < n; i++) {
-			for (int j = 0; j < m; j++) { io.print(grid[i][j]); }
-			io.println();
-		}
-		io.close();
+		return removableRegions;
 	}
 
-	// Returns the size of the region containing square (i, j)
-	static int floodfill(int i, int j, int r, int color) {
-		// Exit if out of bounds, already visited, or if the current square
-		// has a different color
-		if (i < 0 || i >= n || j < 0 || j >= m) return 0;
-		if (region[i][j] != 0) return 0;
-		if (grid[i][j] != color) return 0;
-
-		// If we haven't exited, we can continue with floodfill
-		region[i][j] = r;
-		int size = 1;
-		size += floodfill(i + 1, j, r, color);
-		size += floodfill(i - 1, j, r, color);
-		size += floodfill(i, j + 1, r, color);
-		size += floodfill(i, j - 1, r, color);
-		return size;
+	static void applyGravity(char[][] grid) {
+		int n = grid.length;
+		for (int col = 0; col < MAX_COLS; col++) {
+			int emptyRow = n - 1;  // Start from the bottom of the column
+
+			// Move non-empty cells to the lowest available position in the column
+			for (int row = n - 1; row >= 0; row--) {
+				if (grid[row][col] != '0') {
+					char temp = grid[emptyRow][col];
+					grid[emptyRow][col] = grid[row][col];
+					grid[row][col] = temp;
+					emptyRow--;
+				}
+			}
+		}
 	}
 
-	// CodeSnip{Kattio}
+	public static void main(String[] args) throws IOException {
+		BufferedReader br = new BufferedReader(new FileReader("mooyomooyo.in"));
```

Comment:
use `Kattio CodeSnip` for consistency

Reply:
plus, in contest who tf is copying over kattio